src/sage/rings/polynomial/polynomial_zmod_flint.pyx

```diff
@@ -532,6 +532,16 @@ cdef class Polynomial_zmod_flint(Polynomial_template):
             (3*x^4 + 2*x^3 + x^2 + 2*x, x^4 + 3*x^3 + x^2 + x)
             sage: (p*d % x^9) == n
             True
+
+        Check that :issue:`37169` is fixed - it does not throw an error::
+
+            sage: R.<x> = Zmod(4)[]
+            sage: _.<z> = R.quotient_ring(x^2 - 1)
+            sage: c = 2 * z + 1
+            sage: c * Zmod(2).zero()
+            Traceback (most recent call last):
+            ...
+            RuntimeError: Aborted
```

Comment:
Is this the best error we can give the user?

Reply:
I don't know how to properly catch/handle all erroring cases in Python. I don't even understand why this is erroring right now, so it is probably best to leave this for now. Maybe a better way is for `sig_on()` `sig_off()` to return the runtime exception message or something

Reply:
Multiplication by zero should just be fine though... right? So my guess is there's some case where the flint code ensures the parents are the same and when they're not the code crashes?